tests/plugins/dialog/positioning.js

```diff
@@ -218,8 +221,39 @@
 					} );
 				} );
 			} );
-		}
+		},
+
+		// (#3638)
+		'test dialog is positioned above the cover after opening the same dialog twice': function() {
+			var editorBot = this.editorBot;
 
+			editorBot.dialog( 'link', function() {
+				editorBot.dialog( 'link', function( dialog ) {
+					var cover = CKEDITOR.document.findOne( '.cke_dialog_background_cover' ),
+						coverZIndex = parseInt( cover.getStyle( 'z-index' ), 10 ),
+						dialogZIndex = parseInt( dialog.getElement().getStyle( 'z-index' ), 10 );
+
+					dialog.getButton( 'cancel' ).click();
+
+					assert.isTrue( dialogZIndex > coverZIndex, 'Dialog should be above the cover' );
```

Comment:
We don't have anything similar to [chai isAbove](https://www.chaijs.com/api/assert/#method_isabove), so maybe we could at least give both zIndexes values to the assert message? In case of fail there will be already some results visible besides `expected true but was false`.